syntax = "proto3";

package customers;

service Customers{

rpc GetRowsByColumn(SelectOneMessage) returns (generalResponse){}
rpc GetRowByMultiColumns(SelectManyMessage) returns (generalResponse){}
rpc InsertSeller(InsertMessage) returns (generalResponse){}
rpc UpdateRowByColumn(UpdateMessage) returns (generalResponse){}

}

message InsertMessage{
    string table_name = 1;
    string columns = 2;
    string values = 3;
}
message UpdateMessage{
    string table_name = 1;
    string columns = 2;
    string values = 3;
    string condition_col = 4;
    int32 condition_val = 5;
}

message SelectOneMessage{
    string table_name = 1;
    string column = 2;
    string search_value = 3;
}

message SelectManyMessage{
    string table_name = 1;
    string columns = 2;
    string search_values = 3;
    string selected_columns = 4;
}
message generalResponse{
    string msg=1;
}
